### ISSUE TITLE
questions about the loss.

### ISSUE THREAD

**Issue by WormCoder** [eriklindernoren/PyTorch-YOLOv3#65]
https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/959e0ff43f5b82bdacef87f4240bae8415eac45b/models.py#L162

@eriklindernoren thanks for sharing!
I read the yolov3 paper and yolo_layer's code of Darknet. About the loss, it seems that the code is different with the paper.
In the paper, the author claimed that mse loss was employed for box regression. But in the code(as well as in your Pytorch implementation), bce loss was performed for box center regression. As far as I know, bce loss is for classification with Bernoulli distribution assumption. So why do you use bce loss here?

**Comment by ECer23** [comment]
I think so. I re-implemented the loss function as this repo does, but it seems the loss would blow out. If I change `nn.BCELoss` to `nn.MSELoss` to compute the coordinate loss, the classification loss and confidence loss will decrease while coordinates loss won't change.

Details of my implementation could be seen in [](https://github.com/ECer23/yolov3.pytorch/blob/master/src/layers.py)

And the `lambda` seems not right. I find another implementation of YOLO v2, and the loss function looks different from this implementation.

![](https://lh6.googleusercontent.com/proxy/IUw0XZ2uik13XQBSaYofYaDquiucNqsj5paNobjwVxenbOoE0YoVXeW0Q00F12qfA64zy0DB_8vSfkEdgkBzICONgHKMVbjmPCYfGTuDExtzWSgY=w5000-h5000)

**Comment by eriklindernoren** [comment]
Hi, let's keep training issues in #2.
